baseServer_test.go

```diff
@@ -242,37 +274,42 @@ func TestWaitForMessagingToEnd(t *testing.T) {
 	const numAgents = 10
 	m := make([]basePlatformSOMAS.AgentGeneratorCountPair[ITestBaseAgent], 1)
 	m[0] = basePlatformSOMAS.MakeAgentGeneratorCountPair(NewTestAgent, numAgents)
-	server := basePlatformSOMAS.CreateServer(m, 1, 1, time.Second)
+	server := basePlatformSOMAS.CreateServer(m, 1, 1, time.Second, time.Millisecond)
 	agentMap := server.GetAgentMap()
 	arrayOfIDs := make([]uuid.UUID, numAgents)
 	i := 0
 	for id, ag := range agentMap {
 		arrayOfIDs[i] = id
 		i++
-		ag.SetGoal(numberOfMessages *(numAgents))
+		ag.SetGoal(numberOfMessages * (numAgents))
 	}
 
 	for j := 0; j < numberOfMessages; j++ {
-		for _,ag := range server.GetAgentMap() {
-		msg := ag.NewTestMessage()
-		//server.IncrementWaitGroup()
-		go ag.SendMessage(msg, arrayOfIDs)
-	}}
-	server.EndAgentListeningSession()
+		for _, ag := range server.GetAgentMap() {
+			msg := ag.NewTestMessage()
+			go ag.SendMessage(msg, arrayOfIDs)
+		}
+	}
+	a, b := server.EndAgentListeningSession()
```

Comment:
Would be better to return a boolean, as we discussed before. Then, we can have the error written here, which massively improves readability